README.md

```diff
@@ -1,18 +1,22 @@
 # Selfauth
-self-hosted auth_endpoint using simple login mechanism
 
-# Warning: while Selfauth will work with old versions of PHP, the ability to get sufficiently random strings was not added until version 5.6.  While older versions are not completely insecure, it is strongly recommended you upgrade to a newer version of PHP.
+Selfauth is a self-hosted [Authorization Endpoint](https://indieweb.org/authorization-endpoint) used to login with a personal URL (as [Web sign-in](http://indieweb.org/Web_sign-in)) via [IndieAuth](https://indieweb.org/IndieAuth). See [How it works](#how-it-works) for more.
 
-# Warning: currently selfauth only supports authentication, not authorization.  Meaning scopes will not work yet.  If you need access to these features, it is advisable to use something else for now.
 
-Setup
------
+## Warnings
+
+- While Selfauth will work with old versions of PHP, the ability to get sufficiently random strings was not added until version 5.6. While older versions are not completely insecure, **it is strongly recommended you upgrade to a newer version of PHP**.
+
+- Currently selfauth only supports authentication, not authorization.  Meaning **scopes will not work yet**. If you need access to these features, it is advisable to use something else for now.
```

Comment:
This isn’t strictly true. Selfauth will happily issue a code for authorisation use, with scopes included in the HMAC and code.